src/overlap_checker.cpp

```diff
@@ -177,81 +153,98 @@ are_bboxs_disjoint(const Bnd_OBB &b1, const Bnd_OBB& b2, double tolerance)
 int
 main(int argc, char **argv)
 {
-	configure_spdlog();
+	configure_aixlog();
 
 	std::string path_in;
-	unsigned num_parallel_jobs;
+	unsigned num_parallel_jobs = 4;
 	double
 		bbox_clearance = 0.5,
 		max_common_volume_ratio = 0.01;
 	std::vector<double> imprint_tolerances = {0.001, 0};
 
 	{
-		CLI::App app{"Perform imprinting of BREP shapes."};
-		app.add_option(
-			"input", path_in,
-			"Path of the input file")
-			->required()
-			->option_text("file.brep");
-		app.add_option(
-			"-j", num_parallel_jobs,
-			"Parallelise over N threads")
-			->option_text("N")
-			->default_val(4)
-			->check(CLI::Range(1, 1024));
-		app.add_option(
-			"--bbox-clearance", bbox_clearance,
-			"Bounding-boxes closer than C will be checked for overlaps")
-			->option_text("C");
-		app.add_option(
-			"--imprint-tolerance", imprint_tolerances,
-			"Faces, edges, and verticies will be merged when closer than T")
-			->option_text("T")
-			->expected(1, 10);
-		app.add_option(
-			"--max-common-volume-ratio", max_common_volume_ratio,
-			"Imprinted volume with ratio <R is considered acceptable")
-			->option_text("R")
-			->check(CLI::Range(0., 1.));
-		CLI11_PARSE(app, argc, argv);
-	}
+		const char * doc = "Perform imprinting of BREP shapes.";
+		const char * usage = "input.brep";
+
+
+		std::stringstream stream;
+
+		stream << "Parallelise over N[=" << num_parallel_jobs << "] threads";
+		auto help_num_par_jobs = stream.str();
+		stream = {};
+
+		stream << "Bounding-boxes closer than C[=" << bbox_clearance << "] will be checked for overlaps";
+		auto help_bbox_cl = stream.str();
+		stream = {};
+
+		stream
+			<< "Faces, edges, and verticies will be merged when closer than T[="
+			<< imprint_tolerances[0] << ']';
+		auto help_imp_tol = stream.str();
+		stream = {};
+
+		stream
+			<< "Imprinted volume with ratio <R[="
+			<< max_common_volume_ratio << "] is considered acceptable";
+		auto help_max_common = stream.str();
+		stream = {};
+
+		cxx_argp::parser argp(1);
+		argp.add_option(
+			{"jobs", 'j', "N", 0, help_num_par_jobs.c_str()}, num_parallel_jobs);
+		argp.add_option(
+			{"bbox-clearance", 1024, "C", 0, help_bbox_cl.c_str()}, bbox_clearance);
+		argp.add_option(
+			{"imprint-tolerance", 1025, "T", 0, help_imp_tol.c_str()}, imprint_tolerances);
+		argp.add_option(
+			{"max-common-volume-ratio", 1026, "R", 0, help_max_common.c_str()}, max_common_volume_ratio);
+
+		if (!argp.parse(argc, argv, usage, doc)) {
+			return 1;
+		}
 
-	// make sure parameters are sane!
-	{
+		const auto &args = argp.arguments();
+		assert(args.size() == 1);
+		path_in = args[0];
+
+		if (num_parallel_jobs < 1 || num_parallel_jobs > 1024) {
```

Comment:
why 1024?

Reply:
sanity check mostly, not sure if there are many systems with close to that many cores available.

Reply:
Is it just threads?

Reply:
smells like a magic number, is there no compiler macro to check how many cpus are the on the machine?

Reply:
yes it's a magic number, so should probably be in a macro.  it's just there to stopping the user from shooting themselves in the foot

I look at `max_parallel` later to generate a more appropriate warning later on